### PR TITLE
Fix combination of alias and default

### LINKED ISSUE #353: Aliases are not checked when a default value is provided
When providing a default value, pyserde doesn't check any aliases.

Example:
```python
from dataclasses import dataclass

from serde import field, serde
from serde.json import from_json


@serde
@dataclass
class Foo:
    a: int = field(default=2, alias=["b", "c", "d"])


def main():
    s = '{"a": 10}'
    print(f"From Json: {from_json(Foo, s)}")
    # Expected: From Json: Foo(a=10)
    # Actual: From Json: Foo(a=10)

    s = '{"b": 20}'
    print(f"From Json: {from_json(Foo, s)}")
    # Expected: From Json: Foo(a=20)
    # Actual: From Json: Foo(a=2)


if __name__ == "__main__":
    main()
```

### PR DESCRIPTION
**Pull request by kmsquire**
Fixes #353

### CODE REVIEW

File: serde/de.py [hunk 1]
```diff
@@ -876,8 +884,12 @@ def literal(self, arg: DeField) -> str:
         func_name = literal_func_name(type_args(arg.type))
         return f"serde_scope.funcs['{func_name}'](cls=cls, data={arg.data}, reuse_instances=reuse_instances)"
 
+    # HERE
```
[the remark this answers is not in the document]
Comment (kmsquire) [reply serde/de.py:887]: Nope, sorry, will remove (and rebase).

**Comment by kmsquire** [comment]
Rebased and removed the spurious comment.

**Comment by kmsquire** [comment]
Added a couple of lines to the test code to fix code coverage.  Should be good to go.

File: serde/de.py [hunk 2]
```diff
@@ -878,6 +886,9 @@ def literal(self, arg: DeField) -> str:
 
     def default(self, arg: DeField, code: str) -> str:
         exists = f'"{arg.conv_name()}" in {arg.datavar}'
+        if arg.alias:
+            aliases = map(lambda s: f'"{s}"', [arg.name, *arg.alias])
+            exists += f' or _exists_by_aliases({arg.datavar}, [{",".join(aliases)}])'
```
Comment (yukinarit) [inline serde/de.py:891]: I saw the generated code for the following `Foo` class. 

```python
    @serde.serde
    class Foo:
        a: int = serde.field(default=2, alias=["b", "c", "d"])
```

Here is the code excerpt from `from_dict`.
```python
    __a = (
        (_get_by_aliases(data, ["a", "b", "c", "d"]))
        if "a" in data or _exists_by_aliases(data, ["a", "b", "c", "d"])
        else serde_scope.defaults["a"]
    )
```

I think "a" is checked twice? Maybe we can further optimize to generate the code like this? 🤔  
```python
    __a = (
        (_get_by_aliases(data, ["a", "b", "c", "d"]))
        _exists_by_aliases(data, ["a", "b", "c", "d"])
        else serde_scope.defaults["a"]
    )
```

Comment (yukinarit) [reply serde/de.py:891]: I tested this works 👍 
```python
    def default(self, arg: DeField, code: str) -> str:
        if arg.alias:
            aliases = map(lambda s: f'"{s}"', [arg.name, *arg.alias])
            exists = f'_exists_by_aliases({arg.datavar}, [{",".join(aliases)}])'
        else:
            exists = f'"{arg.conv_name()}" in {arg.datavar}'

```

**Review by yukinarit — COMMENTED** [review]
Thanks, I found 1 more thing, Please check!

File: serde/de.py (hunk 2 above)
Comment (kmsquire) [reply serde/de.py]: Thanks!  I pushed your version.

**Pull request merged**